src/pip/_internal/vcs/git.py

```diff
@@ -155,7 +165,7 @@ def resolve_revision(cls, dest, url, rev_options):
                 rev,
             )
 
-        if not rev.startswith('refs/'):
+        if rev and not rev.startswith('refs/'):
```

Comment:
The way I would handle this is add `assert rev is not None` after the `rev = rev_options.arg_rev` line above (which is true for the `Git` class if you look at the implementation of the `arg_rev` property).